transport/http/handler.go

```diff
@@ -105,9 +105,18 @@ func (h handler) callHandler(w http.ResponseWriter, req *http.Request, start tim
 		ctx = baggage.NewContextWithHeaders(ctx, headers.Items())
 	}
 
-	handler, err := h.Registry.GetHandler(treq.Service, treq.Procedure)
-	if err == nil {
-		err = internal.SafelyCallHandler(handler, start, ctx, httpOptions, treq, newResponseWriter(w))
+	handlerInfo, err := h.Registry.GetHandler(treq.Service, treq.Procedure)
+	if err != nil {
+		return err
+	}
+
+	switch handlerInfo.Mode {
```

Comment:
Let's be consistent when referring to a first-class concept and call this: `spec.RPCMode`.